pkg/ottl/ottlfuncs/func_delete_test.go

```diff
@@ -0,0 +1,377 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+var nilOptional ottl.Optional[ottl.IntGetter[any]]
+
+func setter(_ context.Context, res, val any) error {
+	rSlice := res.(pcommon.Slice)
+	vSlice := val.(pcommon.Slice)
+	return rSlice.FromRaw(vSlice.AsRaw())
+}
+
+func getIntGetter(index int64) ottl.IntGetter[any] {
+	return ottl.StandardIntGetter[any]{
+		Getter: func(_ context.Context, _ any) (any, error) {
+			return index, nil
+		},
+	}
+}
+
+func getSlice(t *testing.T, expected []any) pcommon.Slice {
```

Comment:
I might be missing something, but couldn't we use `FromRaw` to achieve the same?

```go
func asSlice(t *testing.T, expected []any) pcommon.Slice {
	expectedSlice := pcommon.NewSlice()
	err := expectedSlice.FromRaw(expected)
	require.NoError(t, err)
	return expectedSlice
}
```